### PR TITLE
Add python3.12

### PR DESCRIPTION
**Pull request by phackstock**
Looks like the nomenclature python3.12 upgrade is working fine.

### CODE REVIEW

File: docs/installation.rst [hunk 1]
```diff
@@ -5,7 +5,7 @@
 Installation
 ============
 
-.. attention:: The nomenclature requires python >= 3.8
+.. attention:: The nomenclature requires python >= 3.10, < 3.13
```
Comment (danielhuppmann) [inline docs/installation.rst:8]: Shouldn’t we remove this warning entirely?

File: docs/index.rst [hunk 2]
```diff
@@ -13,7 +13,7 @@ Release v\ |version|.
 .. |doi| image:: https://zenodo.org/badge/375724610.svg
    :target: https://zenodo.org/badge/latestdoi/375724610
 
-.. |python| image:: https://img.shields.io/badge/python-_3.10|_3.11-blue?logo=python&logoColor=white
+.. |python| image:: https://img.shields.io/badge/python-3.10_|_3.11_|_3.12-blue?logo=python&logoColor=white
```
Comment (danielhuppmann) [inline docs/index.rst:16]:
```suggestion
.. |python| image:: https://img.shields.io/badge/python-≥3.10,<3.13-blue?logo=python&logoColor=white
```

File: README.md [hunk 3]
```diff
@@ -7,7 +7,7 @@ the [LICENSE](LICENSE) for details.
 
 [![license](https://img.shields.io/badge/License-Apache%202.0-black)](https://github.com/IAMconsortium/nomenclature/blob/main/LICENSE)
 [![DOI](https://zenodo.org/badge/375724610.svg)](https://zenodo.org/badge/latestdoi/375724610)
-[![python](https://img.shields.io/badge/python-3.10_|_3.11-blue?logo=python&logoColor=white)](https://github.com/IAMconsortium/nomenclature)
+[![python](https://img.shields.io/badge/python-3.10_|_3.11_|_3.12-blue?logo=python&logoColor=white)](https://github.com/IAMconsortium/nomenclature)
```
Comment (danielhuppmann) [inline README.md:10]: For consistency with pyam
```suggestion
[![python](https://img.shields.io/badge/python-≥3.10,<3.13-blue?logo=python&logoColor=white)](https://github.com/IAMconsortium/nomenclature)
```

**Review by danielhuppmann — APPROVED** [review]
Thanks! A few nitpicky suggestions inline, good to merged either way

File: docs/installation.rst (hunk 1 above)
Comment (phackstock) [reply docs/installation.rst:8]: Fair point.

**Pull request merged**